packages/p2p/src/components/modal-manager/modals/share-my-ads-modal/share-my-ads-modal.tsx

```diff
@@ -0,0 +1,155 @@
+import React from 'react';
+import { toPng } from 'html-to-image';
+import {
+    Button,
+    Clipboard,
+    DesktopWrapper,
+    Icon,
+    MobileWrapper,
+    Modal,
+    Text,
+    useCopyToClipboard,
+} from '@deriv/components';
+import { useIsMounted } from '@deriv/shared';
+import { observer } from '@deriv/stores';
+import { Localize, localize } from 'Components/i18next';
+import { useModalManagerContext } from 'Components/modal-manager/modal-manager-context';
+import MyProfileSeparatorContainer from 'Components/my-profile/my-profile-separator-container';
+import { TAdvert } from 'Types';
+import ShareMyAdsIcons from './share-my-ads-socials';
+import ShareMyAdsCard from './share-my-ads-card';
+
+const ShareMyAdsModal = ({ advert }: TAdvert) => {
+    const [show_popup, setShowPopup] = React.useState(false);
+    const [is_copied, copyToClipboard, setIsCopied] = useCopyToClipboard();
+
+    const isMounted = useIsMounted();
+    const divRef = React.useRef(null);
+    // TODO: replace with proper url when available
+    const advert_url = window.location.href;
+
+    const { hideModal, is_modal_open } = useModalManagerContext();
+
+    let timeout_clipboard: ReturnType<typeof setTimeout>;
+
+    const onCopy = (event: { stopPropagation: () => void }) => {
+        copyToClipboard(advert_url);
+        timeout_clipboard = setTimeout(() => {
+            if (isMounted()) {
+                setIsCopied(false);
+            }
+        }, 2000);
+        event.stopPropagation();
+    };
+
+    const handleGenerateImage = () => {
+        if (divRef.current) {
+            toPng(divRef.current).then(dataUrl => {
+                const link = document.createElement('a');
+                link.download = 'test.png';
+                link.href = dataUrl;
+                link.click();
+            });
+        }
+    };
+
+    // TODO: Replace with proper message and url when available
+    const handleShareLink = () => {
+        navigator.share({
+            url: advert_url,
+            title: 'P2P Advert',
+            text: 'This is my advert!',
+        });
+    };
+
+    React.useEffect(() => {
+        return () => {
+            clearTimeout(timeout_clipboard);
+        };
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, []);
+
+    return (
+        <React.Fragment>
+            <Modal
+                has_close_icon
+                is_open={is_modal_open}
+                title={localize('Share this ad')}
+                toggleModal={() => (show_popup ? {} : hideModal())}
```

Comment:
```suggestion
                toggleModal={() => {
                    if (show_popup) return;
                    hideModal()
                })}
```
This looks more straightforward than returning an empty object

Reply:
ah i forgot to remove this check since we no longer have the state for show_popup anymore